ctf2ctf.cpp

```diff
@@ -1335,15 +1335,32 @@ struct Event
 
         cpuId = rawCpuId.value();
 
-        tid = context->tid(cpuId);
-        pid = context->pid(tid);
+        if (context->options.ctxPidTid) {
```

Comment:
can you explain why this needs to be optional? i.e. can't we check if the `vpid` and `vtid` values exist, and if so, use those unconditionally? and only keep the `context->tid/pid` calls as a fallback?

Reply:
TBH this was the initial solution but I've decided that it will might bring regressions in other cases, so for sake of safety I went the command line option way to make it more explicit.

Reply:
yes, please remove the option - I don't see why LTTng would lie to us - i.e. if we have the vpid/vtid, we should be able to use it directly. If not, we can try the fallback.